yktrace/src/tir.rs

```diff
@@ -138,11 +179,29 @@ impl TirTrace {
                         let op = if SIR.bodies.contains_key(callee_sym) {
                             // We have SIR for the callee, so it will appear inlined in the trace
                             // and we only need to emit Enter/Leave statements.
-                            TirOp::Statement(Statement::Enter(
-                                op.clone(),
-                                args.to_vec(),
-                                dest.as_ref().map(|(ret_val, _ret_bb)| ret_val.clone())
-                            ))
+
+                            // Rename the destination if there is one.
+                            let newdest = dest.as_ref().map(|(ret_val, _ret_bb)|
+                                TirTrace::rename_place(&rename_map, &ret_val));
+                            // Rename all `Local`s within the arguments.
+                            let mut newargs = Vec::new();
+                            for (i, op) in args.iter().enumerate() {
+                                newargs.push(TirTrace::rename_operand(&rename_map, &op));
+                                let oldlocal = Local(i as u32 + 1);
+                                let newlocal = Local(var_count);
+                                if let Some(v) = rename_map.get_mut(&oldlocal) {
+                                    v.push(newlocal);
+                                } else {
+                                    rename_map.insert(oldlocal, vec![newlocal]);
+                                }
+                                var_count += 1;
+                            }
+                            cur_call_args.push(newargs.len() as u32);
+                            // FIXME It seems that calls always have a destination despite it being
```

Comment:
Should we add an `assert` along the lines of `assert!(call.destination.is_some())` to make sure this assumption can't be violated?

Reply:
Good idea.

Reply:
Fixed in 24eabc1